### PR TITLE
STP-10/module refactor

### PR DESCRIPTION
**Pull request by JoseMotto-sg**
## Change description

> Refactor for tf automation

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
 terraform {
-  required_version = ">= 0.12"
-}
+  required_version = ">= 1.1"
+
+  required_providers {
+    aws = {
+      version = ">= 3.63"
+      source  = "hashicorp/aws"
+      version = ">= 4.9"
```
Comment (DJKaswa) [inline versions.tf]: Did this actually pass with two versions?

File: README.md [hunk 2]
```diff
@@ -28,5 +32,76 @@ module "codebuild_webdeploy_app_name" {
...
   input_tags             = local.common_tags
 }
-
 ```
+## StratusGrid Standards we assume
+- All resource names and name tags shall use `_` and not `-`s
+- The old naming standard for common files such as inputs, outputs, providers, etc was to prefix them with a `-`, this is no longer true as it's not POSIX compliant. Our pre-commit hooks will fail with this old standard.
+- StratusGrid generally follows the TerraForm standards outlined [here](https://www.terraform-best-practices.com/naming)
+## Repo Knowledge
+Repository for Module vmimport
```
Comment (DJKaswa) [inline README.md:41]: I don't think this is the correct module name. See comment on the config file.

File: README.md [hunk 3]
```diff
@@ -1,4 +1,7 @@
-# codebuild-webdeploy-package
+<!-- BEGIN_TF_DOCS -->
+# terraform-aws-iam-role-vmimport
+GitHub: [StratusGrid/terraform-aws-iam-role-vmimport](https://github.com/StratusGrid/terraform-aws-iam-role-vmimport)
+
```
Comment (DJKaswa) [inline README.md]: Seems like a copy/paste mistake here. This repo is not for vmimport.

File: .config/.terraform-docs.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+version: ""
+
+formatter: markdown table
+
+name: terraform-aws-iam-role-vmimport
+
+content: |-
+  # terraform-aws-iam-role-vmimport
+  GitHub: [StratusGrid/terraform-aws-iam-role-vmimport](https://github.com/StratusGrid/terraform-aws-iam-role-vmimport)
+
```
Comment (DJKaswa) [inline .config/.terraform-docs.yml]: Check the name here. There seems to be a copy/paste error on the name of this module.

**Review by DJKaswa — DISMISSED** [review]
Please change the name of the module in the config file, which should generate a new Readme. Please see individual comments.

File: versions.tf (hunk 1 above)
Comment (JoseMotto-sg) [reply versions.tf]: Yeah because it passes the conditions, but I deleted the old one

File: README.md (hunk 2 above)
Comment (JoseMotto-sg) [reply README.md:41]: Fixed

File: README.md (hunk 3 above)
Comment (JoseMotto-sg) [reply README.md]: Fixed

File: .config/.terraform-docs.yml (hunk 4 above)
Comment (JoseMotto-sg) [reply .config/.terraform-docs.yml]: Fixed

**Review by chrischildresssg — APPROVED** [review]
Approved

**Review by chrischildresssg — DISMISSED** [review]
Approved

**Review by chrischildresssg — APPROVED** [review]
Approved

**Pull request merged**